radio/src/telemetry/flysky_ibus.cpp

```diff
@@ -0,0 +1,176 @@
+  // External voltage
+  {0x03,            ZSTR_A3,                UNIT_VOLTS,                  1},
+  // RX Noise
+  {0xfb,            ZSTR_RX_SNR,            UNIT_DB,                     0},
+  // RX RSSI
+  {0xfc,            ZSTR_RSSI,              UNIT_DB,                     0},
+  // RX error rate
+  {0xfe,            ZSTR_RX_QUALITY,        UNIT_RAW,                    1},
+  // 0xff is an used sensor slot)
+  // Pseud sensor for TRSSI
+  {TX_RSSI_ID,      ZSTR_TX_RSSI,           UNIT_RAW,                     0},
+  // sentinel
+  {0x00,            NULL,                   UNIT_RAW,                    0},
+
+
+};
+
+static void processFlySkySensor(const uint8_t *packet) {
+  uint16_t id = packet[0];
+  const uint8_t instance = packet[1];
+  const int32_t value = (packet[3] << 8)  + packet[2];
+
+  if (id == 0xff)
+    // No sensor
+    return;
+  if (id == 0)
+    // Some part of OpenTX does not like sensor with id and instance 0, remap to 0x100
+    id = 0x100;
+
+  if(id == 0xfe)
+    telemetryData.rssi.set(value);
+
+  for (const FlySkySensor * sensor = flySkySensors; sensor->id; sensor++) {
+    // Extract value, skip header
+
+    setTelemetryValue(TELEM_PROTO_FLYSKY_IBUS, id, 0, instance, value, sensor->unit, sensor->precision);
+    return;
+  }
+  setTelemetryValue(TELEM_PROTO_FLYSKY_IBUS, id, 0, instance, value, UNIT_RAW, 0);
+}
+
+static void processFlySkyPacket(const uint8_t *packet) {
+  // Set TX RSSI Value, reverse MULTIs scaling
+  setTelemetryValue(TELEM_PROTO_FLYSKY_IBUS, TX_RSSI_ID, 0, 0, packet[1], UNIT_RAW, 0);
+
+  for (int sensor = 0; sensor < 7; sensor++) {
+    int index = 2 + (4 * sensor);
+    processFlySkySensor(packet+index);
+  }
+  telemetryStreaming = TELEMETRY_TIMEOUT10ms;
+}
+
+
+void processFlySkyTelemetryData(uint8_t data)
+{
+  if (telemetryRxBufferCount == 0 && data != 0xAA) {
+    TRACE("[IBUS] invalid start byte 0x%02X", data);
+    return;
+  }
+
+  if (telemetryRxBufferCount < TELEMETRY_RX_PACKET_SIZE) {
+    telemetryRxBuffer[telemetryRxBufferCount++] = data;
+  }
+  else {
+    TRACE("[IBUS] array size %d error", telemetryRxBufferCount);
+    telemetryRxBufferCount = 0;
+  }
+
+
+  if (telemetryRxBufferCount >= FLYSKY_TELEMETRY_LENGTH) {
+    // debug print the content of the packets
+#if 0
+    debugPrintf("[IBUS] Packet 0x%02X rssi 0x%02X: ",
+                telemetryRxBuffer[0], telemetryRxBuffer[1]);
+    for (int i=0; i<7; i++) {
+      debugPrintf("[%02X %02X %02X%02X] ", telemetryRxBuffer[i*4+2], telemetryRxBuffer[i*4 + 3],
+                  telemetryRxBuffer[i*4 + 4], telemetryRxBuffer[i*4 + 5]);
+    }
+    debugPrintf("\r\n");
+#endif
+    processFlySkyPacket(telemetryRxBuffer);
+    telemetryRxBufferCount = 0;
+  }
+}
+
+const FlySkySensor *getFlySkySensor(uint16_t id)
+{
+  for (const FlySkySensor * sensor = flySkySensors; sensor->id; sensor++) {
+    if (id == sensor->id)
+      return sensor;
+  }
+  return nullptr;
```

Comment:
The microsoft site says it is support since at least VC 2010 (and that version probably does not like QT5):
https://msdn.microsoft.com/en-us//library/hh567368.aspx